### PR TITLE
[node] Allow undefined being passed as the error parameter

### PR DESCRIPTION
**Pull request by LinusU**
This brings the changed signatures in line with the signatures that doesn't have a `TResult`.

Without this you cannot properly type promisified functions that returns `undefined` instead of `null` when there isn't an error, e.g. `getLength` on `form-data`. 

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/nodejs/node/blob/08977822ee40b69e424b09bdb5f30fb3edbe23a8/lib/internal/util.js#L287
- [n/a] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [n/a] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

File: types/node/v9/base.d.ts [hunk 1]
```diff
@@ -6087,17 +6087,42 @@ declare module "util" {
     export function callbackify<T1, T2, T3, T4, T5, T6, TResult>(fn: (arg1: T1, arg2: T2, arg3: T3, arg4: T4, arg5: T5, arg6: T6) => Promise<TResult>): (arg1: T1, arg2: T2, arg3: T3, arg4: T4, arg5: T5, arg6: T6, callback: (err: NodeJS.ErrnoException, result: TResult) => void) => void;
```
Comment (SimonSchick) [inline types/node/v9/base.d.ts]: Please don't update EOL type definitions.

**Review by SimonSchick — CHANGES_REQUESTED** [review]
I'd suggest making the callback it's own type, the duplication in `promisify` definitions is a bit high.

**Comment by LinusU** [comment]
> I'd suggest making the callback it's own type

Would you mind elaborating on this one? The callback will be different for each row, right? 🤔 or did you mean to make a type out of `Error | null | undefined`? 🤔

**Comment by LinusU** [comment]
@SimonSchick 

- Removed changes to EOL node
- Used specific types instead of repeating
- Fixed too long lines (don't know how this slipped in in the beginning 🤔)
- Rebased on `master`

🚀

**Comment by LinusU** [comment]
ping @SimonSchick ☺️

**Comment by LinusU** [comment]
@SimonSchick friendly nudge ☺️

File: types/node/v10/util.d.ts [hunk 2]
```diff
@@ -87,24 +87,27 @@ declare module "util" {
         fn: (arg1: T1, arg2: T2, arg3: T3, arg4: T4, arg5: T5, arg6: T6) => Promise<TResult>
     ): (arg1: T1, arg2: T2, arg3: T3, arg4: T4, arg5: T5, arg6: T6, callback: (err: NodeJS.ErrnoException | null, result: TResult) => void) => void;
 
+    type Errback = (err?: Error | null) => void;
```
Comment (SimonSchick) [inline types/node/v10/util.d.ts]: Can we perhaps rename this to `Callback` and the other one `ValueCallback`?
`Errback` seems like slang.

Comment (smac89) [inline types/node/v10/util.d.ts]: Why is this a union type with null?

```typescript
type Errback = (err?: Error) => void;
```

Comment (LinusU) [reply types/node/v10/util.d.ts]: I thought that `?` implied `| undefined`, not `| null`, I could be wrong though... 🤔

Comment (LinusU) [reply types/node/v10/util.d.ts]: Sure 👍

Comment (sindresorhus) [reply types/node/v10/util.d.ts]: Strictly, speaking, it indicates optionality, but yes, it's `undefined`, not `null`.

**Review by sindresorhus — APPROVED** [review]

**Review by BrunoScheufler — APPROVED** [review]

**Comment by sindresorhus** [comment]
Can someone merge this?

**Comment by trivikr** [comment]
"Revision needed" label was added post review from @SimonSchick on Aug 2
Is this good to go Simon?

**Comment by ExE-Boss** [comment]
This needs rebasing on top of <https://github.com/DefinitelyTyped/DefinitelyTyped/pull/42154>.

**Comment by LinusU** [comment]
Sorry, haven't had time to work on this, hopefully I can pick this up and rebase soon...

**Pull request closed without merging**